packages/teams-js/test/public/meeting.spec.ts

```diff
@@ -23,142 +23,228 @@ describe('meeting', () => {
+        const callbackId = toggleIncomingClientAudioMessage.id;
+        desktopPlatformMock.respondToMessage({
+          data: {
+            id: callbackId,
+            args: [{ errorCode: ErrorCode.INTERNAL_ERROR }, null],
+          },
+        } as DOMMessageEvent);
+        await expect(promise).rejects.toEqual({ errorCode: ErrorCode.INTERNAL_ERROR });
+      });
     });
   });
 
   describe('getIncomingClientAudioState', () => {
+    const allowedContexts = [FrameContexts.sidePanel, FrameContexts.meetingStage];
     it('should not allow calls before initialization', () => {
       expect(() => meeting.getIncomingClientAudioState()).toThrowError('The library has not yet been initialized');
     });
 
-    it('should successfully get the incoming client audio state', async () => {
-      await desktopPlatformMock.initializeWithContext('sidePanel');
-
-      const promise = meeting.getIncomingClientAudioState();
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContext => allowedContext === context)) {
+        return;
+      }
+      it(`should not allow meeting.getIncomingClientAudioState calls from ${context} context`, async () => {
+        expect.assertions(1);
+        await utils.initializeWithContext(context);
 
-      const getIncomingClientAudioMessage = desktopPlatformMock.findMessageByFunc('getIncomingClientAudioState');
-      expect(getIncomingClientAudioMessage).not.toBeNull();
-      const callbackId = getIncomingClientAudioMessage.id;
-      desktopPlatformMock.respondToMessage({
-        data: {
-          id: callbackId,
-          args: [null, true],
-        },
-      } as DOMMessageEvent);
-      await expect(promise).resolves.toBe(true);
+        expect(() => meeting.getIncomingClientAudioState()).toThrowError(
+          `This call is only allowed in following contexts: ${JSON.stringify(
+            allowedContexts,
+          )}. Current context: "${context}".`,
+        );
+      });
     });
 
-    it('should return error code 500', async () => {
-      await desktopPlatformMock.initializeWithContext('meetingStage');
-
-      const promise = meeting.getIncomingClientAudioState();
-
-      const getIncomingClientAudioMessage = desktopPlatformMock.findMessageByFunc('getIncomingClientAudioState');
-      expect(getIncomingClientAudioMessage).not.toBeNull();
-      const callbackId = getIncomingClientAudioMessage.id;
-      desktopPlatformMock.respondToMessage({
-        data: {
-          id: callbackId,
-          args: [{ errorCode: ErrorCode.INTERNAL_ERROR }, null],
-        },
-      } as DOMMessageEvent);
-      await expect(promise).rejects.toEqual({ errorCode: ErrorCode.INTERNAL_ERROR });
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContext => allowedContext === context)) {
+        return;
+      }
+      it(`should successfully send the getIncomingClientAudio message. context: ${context}`, async () => {
```

Comment:
Add specific platform name in the test case as it'll be helpful while debugging.

Reply:
I have not added both platforms testing as part of this PR. This was only for fixing the async/await and to add left out unit tests (only for the platform that we were testing here.).